pkg/ipcache/ipcache.go

```diff
@@ -482,6 +510,27 @@ func (ipc *IPCache) RemoveLabels(cidr netip.Prefix, lbls labels.Labels, resource
 	ipc.RemoveMetadata(cidr, resource, lbls)
 }
 
+// UpsertIdentity upserts a given IP and an associated identity and labels
+// into the ipcache metadata map. This will override any other labels associated
+// with this prefix. Only one resource may provide an identity override.
+// If multiple resources upsert a predefined identity, a warning is emitted and
+// random identity will be picked.
+// If the numeric identity does not match the identity for the given
+// lbls set, a warning is emitted and the numeric identity here is ignored.
+// The given labels are not modified nor is its reference saved, as they're
+// copied when inserting into the map.
+// This will trigger asynchronous calculation of any local identity changes
+// that must occur to associate the specified labels with the prefix, and push
+// any datapath updates necessary to implement the logic associated with the
+// metadata currently associated with the 'prefix'.
```

Comment:
For what it's worth, I think that the current code does handle this case gracefully, it will just log an extraneous warning that the user can't do much about. And I guess at least right now, there are no callers of this interface so there's no way to trigger this condition so it's not exactly a big deal. But it would be nice to make this expectation super clear for future callers.

If we used this from the endpoint logic, then the endpoint itself would already retain a reference to the identity so I think that case would be fine. Not sure about other users that directly allocate an identity.